libbeat/outputs/elasticsearch/config.go

```diff
@@ -24,9 +24,12 @@ type elasticsearchConfig struct {
 }
 
 type Template struct {
-	Name      string `config:"name"`
-	Path      string `config:"path"`
-	Overwrite bool   `config:"overwrite"`
+	Enabled           bool   `config:"enabled"`
+	Name              string `config:"name"`
+	IgnoreES2xVersion bool   `config:"ignore_es2x_version"`
+	Path              string `config:"path"`
+	PathES2x          string `config:"path_es2x"`
+	Overwrite         bool   `config:"overwrite"`
```

Comment:
how about versioned configs like.

In config file:

```
template.versions:
    - version: '2.*' # match all ES 2.x.x versions
      path: ...

template.name: ...
template.path: ...
template.overwrite: ...
```

If any of `versions.<idx>.version` matches the version we discovered during ping, `version.<idx>.path` will be used, otherwise default template file via `template.path` will be used.